src/lib/profiles/data-management/Current/TraitData.cpp

```diff
@@ -88,6 +88,66 @@ WEAVE_ERROR UpdateDictionaryDirtyPathCut::CutPath (PropertyPathHandle aPathhandl
     return err;
 }
 
+WEAVE_ERROR TraitSchemaEngine::MapPathToHandle(const char * aPathString, PropertyPathHandle & aHandle) const
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+    PropertyPathHandle childProperty, curProperty;
+    nl::Weave::TLV::TLVType dummyContainerType = kTLVType_Path;
+    char *token = NULL;
+    char *pathStringCopy = (char *)malloc(strlen(aPathString) + 1);
+    VerifyOrExit(pathStringCopy != NULL, err = WEAVE_ERROR_NO_MEMORY);
+
+    strcpy(pathStringCopy, aPathString);
+    token = strtok(pathStringCopy, "/");
+
+    // initialize the out argument to NULL
+    aHandle = kNullPropertyPathHandle;
+
+    // Set our starting point for traversal to the root node.
+    curProperty = kRootPropertyPathHandle;
+
+    // Descend into our schema tree using the tags encountered to help navigate through the various branches.
+    while (token!= NULL)
+    {
+        // Todo: currently it does not support dictionary
+        const uint64_t tag = atoi(token) | kSpecialTagMarker;
```

Comment:
Must handle case where element is not an integer (e.g. contains non-decimal characters).

Also, you must validate the range of the returned integer.  Context tag number can only be 0 - 255.

Reply:
thanks, done.